File: Ahorn/entities/enemy.jl
module GameHelperEnemy

using ..Ahorn, Maple

@mapdef Entity "GameHelper/Enemy" Enemy(
    x::Integer, y::Integer, speedX::Number=50.0, speedY::Number=200.0,
    hitboxWidth::Number=8.0, hitboxHeight::Number=16.0,
    hitboxXOffset::Number=-3.0, hitboxYOffset::Number=0.0,
    bounceHitboxWidth::Number=8.0, bounceHitboxHeight=4.0,
    bounceHitboxXOffset::Number=-3.0, bounceHitboxYOffset::Number=-3.0,
    fallSpeedLimit::Number=400.0, canDie::Bool=true, drawOutline::Bool=false,
    customSpritePath::String="objects/GameHelper/Enemy"
)

const placements = Ahorn.PlacementDict(
    "Enemy (GameHelper)" => Ahorn.EntityPlacement(
        Enemy,
        "rectangle"
    ),
)

function Ahorn.selection(entity::Enemy)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x, y, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Enemy, room::Maple.Room)
customSpritePath = get(entity.data,"customSpritePath", "objects/GameHelper/Enemy")
if customSpritePath == ""
    customSpritePath = "objects/GameHelper/Enemy"
end
Ahorn.drawSprite(ctx, "$(customSpritePath)/walking00.png", 16, 8)
end

end

File: Loenn/entities/enemy.lua
local enemy = {}

enemy.name = "GameHelper/Enemy"
enemy.depth = 8998
enemy.texture = "objects/GameHelper/Enemy/walking00"
enemy.justification = {0.5, 0.5}
enemy.placements = {
    name = "enemy",
    data = {
        speedX = 50.0,
        speedY = 200.0,
        hitboxWidth = 8.0,
        hitboxHeight = 16.0,
        hitboxXOffset = -3.0,
        hitboxYOffset = 0.0,
        bounceHitboxWidth = 8.0,
        bounceHitboxHeight = 4.0,
        bounceHitboxXOffset = -3.0,
        bounceHitboxYOffset = -3.0,
        canDie = true,
        drawOutline = false,
        customSpritePath = ""
    }
}

return enemy